lib/cliapp/src/Obelisk/CliApp/Process.hs

```diff
@@ -71,8 +71,13 @@ data ProcessSpec = ProcessSpec
 proc :: FilePath -> [String] -> ProcessSpec
 proc cmd args = ProcessSpec (Process.proc cmd args) Nothing
 
+-- | We do not use Process.shell because it invokes "/bin/sh", which
+-- is a huge impurity for us. Unfortunately, we cannot guarantee that
+-- our dependent executables, such as ghcid, do the same.
+--
+-- Instead, we invoke "/usr/bin/env sh -c <cmd>"
 shell :: String -> ProcessSpec
-shell cmd = ProcessSpec (Process.shell cmd) Nothing
+shell cmd = ProcessSpec (Process.proc "/usr/bin/env" ["sh", "-c", cmd]) Nothing
```

Comment:
I don't understand this change. `/usr/bin/env sh` will be the first `sh` in the user's `$PATH`, which is just as of an unknown as whatever `/bin/sh` is — maybe even _more_ of an unknown. If we want this to be a shell _with guaranteed known behaviour_, shouldn't we [do something like this](https://github.com/obsidiansystems/cli-nix/pull/2/commits/3977c891720d9438c3050528cda7a96ec4227f44)?

Reply:
:woman_facepalming: Yea

Reply:
I tested it "on my machine" and thought it worked, but it didn't... because I have bash as my global sh.